strawberryfields/devicespec.py

```diff
@@ -90,13 +90,13 @@ def gate_parameters(self) -> Mapping[str, Ranges]:
         {'squeezing_amplitude_0': x=0, x=1, 'phase_0': x=0, 0≤x≤6.283185307179586}
         """
         gate_parameters = {}
+        if self._spec["gate_parameters"]:
+            for gate_name, param_ranges in self._spec["gate_parameters"].items():
+                # convert gate parameter allowed ranges to Range objects
+                range_list = [[i] if not isinstance(i, Sequence) else i for i in param_ranges]
+                gate_parameters[gate_name] = Ranges(*range_list)
 
-        for gate_name, param_ranges in self._spec["gate_parameters"].items():
-            # convert gate parameter allowed ranges to Range objects
-            range_list = [[i] if not isinstance(i, Sequence) else i for i in param_ranges]
-            gate_parameters[gate_name] = Ranges(*range_list)
-
-        return gate_parameters
+        return gate_parameters or None
```

Comment:
What should be the expected behaviour when the gate parameters dictionary is empty (e.g., a device has a layout with no parameters)? Shouldn't we just return an empty dictionary in that case?

Reply:
That probably makes sense, yes. I'm not sure it matters that much though, or maybe it does. 🤔

Reply:
I think it's worth making the distinction because if the gate parameters are `{}` we know the device is not parameterized; however, if the gate parameters are `None` then we're not sure if the device is not parameterized OR it simply forgot to supply the gate parameters in its specification.